app/components/ui/inline/Button/Button.tsx

```diff
@@ -40,21 +44,35 @@ export const Button = ({
     styles[`button--${size}`],
     styles[`button--${variant}`],
     styles[`button--arrow-${arrowVariant}`],
-
     {
       [styles["mobile-full-width"]]: mobileFullWidth,
       [`${styles["button--link"]} ${styles[`button--link-${variant}`]}`]: href,
     }
   );
 
+  const content = () => {
+    const iconClass = classNames(
+      `fas fa-${iconName}`,
+      styles[`icon-${iconVariant}`]
+    );
+
+    return (
+      <>
+        {iconName && iconVariant === "before" && <span className={iconClass} />}
+        {children}
+        {iconName && iconVariant === "after" && <span className={iconClass} />}
+      </>
```

Comment:
Seems like there would be a handful of outcomes rendered here depending on the arguments being passed in since both `iconName` and `iconVariant` are optional. But I suspect that doesn't matter, it's up to the _user_ to figure out what they want to do?

Reply:
Yes, outcomes are:
Icon left side (if iconName is included)
No icon, default button
Icon right side (if iconName is included)

I had been relying on ::before and ::after css icon declarations but I thought it made sense to just make it reusable